dot/parachain/collator-protocol/message.go

```diff
@@ -155,9 +157,60 @@ func decodeCollationMessage(in []byte) (network.NotificationsMessage, error) {
 	return &collationMessage, nil
 }
 
-func handleCollationMessage(_ peer.ID, msg network.NotificationsMessage) (bool, error) {
-	// TODO: Add things
-	fmt.Println("We got a collation message", msg)
+type ProspectiveCandidate struct {
+	CandidateHash      parachaintypes.CandidateHash
+	ParentHeadDataHash common.Hash
+}
+
+type CollationStatus int
+
+const (
+	// We are waiting for a collation to be advertised to us.
+	Waiting CollationStatus = iota
+	// We are currently fetching a collation.
+	Fetching
+	// We are waiting that a collation is being validated.
+	WaitingOnValidation
+	// We have seconded a collation.
+	Seconded
+)
+
+func (cpvs CollatorProtocolValidatorSide) handleCollationMessage(
+	sender peer.ID, msg network.NotificationsMessage) (bool, error) {
+	if msg.Type() != network.CollationMsgType {
+		return false, fmt.Errorf("unexpected message type, expected: %d, found:%d",
+			network.CollationMsgType, msg.Type())
+	}
+
+	collatorProtocol, ok := msg.(*CollationProtocol)
+	if !ok {
+		return false, errors.New("failed to cast into collator protocol message")
```

Comment:
I would suggest to add the received type in the error
```suggestion
		return false, fmt.Errorf("failed to cast into collator protocol message, expected: *CollationProtocol, got: %T", msg)
```